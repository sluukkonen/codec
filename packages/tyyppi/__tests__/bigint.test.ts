import { bigint } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

test("should parse a bigint", () => {
  expectParseSuccess(bigint, 0n)
})

test("should fail to parse other values", () => {
  expectParseFailure(bigint, 0)
})
