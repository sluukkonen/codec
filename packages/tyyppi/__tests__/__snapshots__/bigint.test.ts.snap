// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should fail to parse other values 1`] = `
[
  {
    "code": "invalid_type",
    "expected": "bigint",
    "message": "Expected a bigint, received a number",
    "path": [],
    "received": "number",
  },
]
`;
